e2e/12-broadcast.spec.js

```diff
@@ -5,122 +5,144 @@ const OTP = require('otp.js');
 const GA = OTP.googleAuthenticator;
 const { takeScreenshot } = require('./helpers/screenshot');
 const { logout, navigateToLogin, login, tapBack, sleep } = require('./helpers/app');
+const { searchRoom } = require('./helpers/roomListView');
 const data = require('./data');
 
+// Tests randomly fail on Android platform - they can fail on:
+// `before`, `should create broadcast room`, `should send message`,
+// `should login as user without write message authorization and enter room`
+// But running tests separately works ok.
 describe('Broadcast room', () => {
-	before(async() => {
-		await device.reloadReactNative();
-	});
+	const broadcastRoom = `broadcast${ data.random }`;
 
-	it('should create broadcast room', async() => {
-		await element(by.id('rooms-list-view-create-channel')).tap();
-		await waitFor(element(by.id('new-message-view'))).toBeVisible().withTimeout(2000);
-		await element(by.id('new-message-view-create-channel')).tap();
-		await waitFor(element(by.id('select-users-view'))).toBeVisible().withTimeout(2000);
-		await element(by.id(`select-users-view-item-${ data.alternateUser }`)).tap();
-		await waitFor(element(by.id(`selected-user-${ data.alternateUser }`))).toBeVisible().withTimeout(5000);
-		await element(by.id('selected-users-view-submit')).tap();
-		await waitFor(element(by.id('create-channel-view'))).toBeVisible().withTimeout(5000);
-		await element(by.id('create-channel-name')).replaceText(`broadcast${ data.random }`);
-		await element(by.id('create-channel-broadcast')).tap();
-		await element(by.id('create-channel-submit')).tap();
-		await waitFor(element(by.id('room-view'))).toBeVisible().withTimeout(60000);
-		await expect(element(by.id('room-view'))).toBeVisible();
-		await waitFor(element(by.text(`broadcast${ data.random }`))).toExist().withTimeout(60000);
-		await expect(element(by.text(`broadcast${ data.random }`))).toExist();
-		await element(by.id('room-view-header-actions')).tap();
-		await waitFor(element(by.id('room-actions-view'))).toBeVisible().withTimeout(5000);
-		await element(by.id('room-actions-info')).tap();
-		await waitFor(element(by.id('room-info-view'))).toBeVisible().withTimeout(2000);
-		await waitFor(element(by.id('room-info-view-broadcast'))).toBeVisible().withTimeout(2000);
-		await expect(element(by.id('room-info-view-broadcast'))).toBeVisible();
-		await tapBack();
-		await waitFor(element(by.id('room-actions-view'))).toBeVisible().withTimeout(2000);
-		await tapBack();
-		await waitFor(element(by.id('room-view'))).toBeVisible().withTimeout(2000);
-		// await tapBack();
-		// await waitFor(element(by.id('rooms-list-view'))).toBeVisible().withTimeout(2000);
-		// await waitFor(element(by.id(`rooms-list-view-item-broadcast${ data.random }`))).toExist().withTimeout(60000);
-		// await expect(element(by.id(`rooms-list-view-item-broadcast${ data.random }`))).toExist();
-	});
+		before(async () => {
+			if (device.getPlatform() === 'ios') {
```

Comment:
Can we do a "global" `if iOS return` instead of one condition per test?

Reply:
unfortunately no :( because detox's object `device` is undefined if I inside of it `describe` block